src/core/thread/link_quality.cpp

```diff
@@ -48,6 +48,18 @@ static const char *const kDigitsString[8] =
     "0",     "125", "25",  "375", "5",   "625", "75",  "875"
 };
 
+void SuccessRateTracker::AddSample(bool aSuccess, uint16_t aWeight)
+{
+    uint32_t newAverage = mFailureRate;
+    uint32_t newValue = (aSuccess) ? 0 : kMaxRateValue;
+
+    // `aWeight/2` is added to the sum to ensure rounding the value to the nearest integer (e.g., 1.1 -> 1, 3.5 -> 4).
+
+    newAverage = ((newAverage * (aWeight - 1)) + newValue + (aWeight / 2)) / aWeight;
```

Comment:
Wouldn't this read more clearly if you used mFailureRate (instead of newAverage) on the RHS of the equation?

Reply:
Good idea. will change.